DevX newsletter: June edition

bobheadxi opened the pull request:
https://deploy-preview-3884--sourcegraph-handbook.netlify.app/departments/engineering/dev/enablement/dev-experience/newsletter/

bobheadxi committed:
init newsletter

jhchabran committed:
Mention the removal of explicit sentry capturing

jhchabran committed:
Add a line about error sampling

bobheadxi committed:
add CLA bot automation update

bobheadxi committed:
retroactively update sourcegraph/log url

bobheadxi committed:
Merge branch 'main' into devx-newsletter-jun

bobheadxi committed:
set a target date for this issue

jhchabran committed:
Log sg start output improvement

jhchabran commented on content/departments/engineering/dev/enablement/dev-experience/newsletter.md:
```diff
@@ -8,6 +8,50 @@ To learn more about components of Sourcegraph's developer experience, check out
+
+### `sg` goodies 
+
+**`sg [cmd...] --feedback`**: TODO @burmudar
```
cc @burmudar !

jhchabran committed:
Add more stuff related to `sg`

jhchabran commented on content/departments/engineering/dev/enablement/dev-experience/newsletter.md:
```diff
@@ -8,6 +8,62 @@ To learn more about components of Sourcegraph's developer experience, check out
+
+**Generated `sg` documentation**: Because maintaining documentation is always hard, what's better than automation to make sure its stays up to date? The [`sg` reference](https://docs.sourcegraph.com/dev/background-information/sg/reference) is now automatically generated. 
+
+**`dev/schemadoc`** has been removed and is replaced by `sg migration ...` commands. Example: `sg migration describe -db codeintel --format=psql -force -out internal/database/schema.codeintel.md` to generate the schema for the `codeintel` db thanks to [#35905](https://github.com/sourcegraph/sourcegraph/pull/35905).
```
cc @efritz if you want to edit/add something there :)

jhchabran committed:
Mention the tech radar for this month.

jhchabran commented on content/departments/engineering/dev/enablement/dev-experience/newsletter.md:
```diff
@@ -8,6 +8,66 @@ To learn more about components of Sourcegraph's developer experience, check out
+
+The linter job that runs on every build is now inferring which linter task needs to run depending on the changes (except `main` where it runs everything), saving some time on pull requests thanks to [#35331](https://github.com/sourcegraph/sourcegraph/pull/35331).
+
+### Tech Radar
```
I think we'll give more visibility and engagement with the radar if we mention it every month. Ideally, it'd be great if team would update it so we'd see the _new_ tags, but that's good enough as it is. cc @vrto

vrto replied:
Good idea, thanks @jhchabran

jhchabran committed:
squash! Prettier

burmudar committed:
add more dev newsletter content

- `--feedback` blurb
- verbose flag when retrying on `sg lint`
- force ready for review message flag

burmudar committed:
squash! Prettier

bobheadxi committed:
slgiht tweaks

bobheadxi committed:
adjust sentry errors section

bobheadxi committed:
add sg analytics entry

bobheadxi committed:
sg setup, sg lint rewrite + auto fixing

bobheadxi commented on content/departments/engineering/dev/enablement/dev-experience/newsletter.md:
```diff
@@ -8,6 +8,83 @@ To learn more about components of Sourcegraph's developer experience, check out
+
+**Generated `sg` documentation**: Because maintaining documentation is always hard, what's better than automation to make sure its stays up to date? The [`sg` reference](https://docs.sourcegraph.com/dev/background-information/sg/reference) is now automatically generated.
+
+**`sg setup` and `sg lint` rewrite**: We have rewritten `sg lint` and `sg setup` to use a shared internal framework that can be used to easily and flexibly build powerful "check-and-fix" tasks. This enables features like:
+
+- `sg setup -check` to quickly generate a report of what you have set up
+- `sg setup -fix` to fix all issues with your dev setup in one go
+- `sg lint -fix` to automatically try and fix your lint issues (only supported by a few linters at the moment, but more can easily be added!)
```
Relies on https://github.com/sourcegraph/sourcegraph/pull/37528

bobheadxi committed:
add note about CI for sg setup

jhchabran commented:
@bobheadxi don't forget to add the loom video from https://sourcegraph.slack.com/archives/C01N83PS4TU/p1655998549750479 !

bobheadxi committed:
add investigating flakes loom

bobheadxi committed:
update newsletter

bobheadxi committed:
Merge branch 'main' of github.com:sourcegraph/handbook into devx-newsletter-jun

bobheadxi committed:
add note about team merge

bobheadxi committed:
squash! Prettier

bobheadxi committed:
fix link
